src/FluentValidation/Validators/EmptyValidator.cs

```diff
@@ -30,13 +30,20 @@ public class EmptyValidator<T,TProperty> : PropertyValidator<T,TProperty> {
 	public override string Name => "EmptyValidator";
 
 	public override bool IsValid(ValidationContext<T> context, TProperty value) {
-		switch (value) {
-			case null:
-			case string s when string.IsNullOrWhiteSpace(s):
-			case ICollection {Count: 0}:
-			case Array {Length: 0}:
-			case IEnumerable e when !e.GetEnumerator().MoveNext():
-				return true;
+		if (value == null) {
+			return true;
+		}
+
+		if (value is string s) {

```

Comment:
@JeremySkinner @ProfFrugal 

Don't these three also change the behavior of the switch expression above.

Line 35 says: "When the value is a string and is empty, return true, otherwise return the EqualityComparer on line 42/49"

This now says: "When the value is a string, return if the string is empty"

The same for the other 2 ICollection and IEnumerable

To maintain same behavior I suggest

```
if (value is string s && string.IsNullOrWhiteSpace(s))
   return true;
\\etc
```

Reply:
@billrob yes you're right - I didn't catch that. I've pushed out an 11.8.1 release with a fix.